src/hazardous/kdf/argon2i.rs

```diff
@@ -58,6 +58,9 @@
 //! - The minimum recommended length for a hashed password is `16` bytes.
 //! - The minimum recommended iteration count is `3`.
 //! - Password hashes should always be compared in constant-time.
+//! - Please note that when verifying, a copy of the computed password hash is placed into
+//! `dst_out`. If the derived hash is considered sensitive in the given context, the user is
+//! responsible for zeroing out this buffer (see [`zeroize crate`]).
```

Comment:
Small formatting nit: I think `[\`zeroize crate\`]` should be `[\`zeroize\` crate]`

Reply:
## Proposal
Consider changing the message slightly to:
> If the derived hash is considered sensitive and you want to provide defense in depth against an attacker reading your application's private memory, the you as the user are responsible for zeroing out this buffer (see [`zeroize crate`]).

## Justification

There are two necessary conditions for an orion user to care about the information in the `dst_out` buffer.

1. The information has to be considered sensitive.
2. The `dst_out` buffer has to be considered vulnerable to compromise.

For user password-hashing applications, it's reasonable to consider the information sensitive. Even though this is for a secure password hash, the best hash is the one that an attacker doesn't have. Also, though it's unlikely that an attacker will gain access to the private memory of the application and be able to actually read the `dst_out` buffer, it does happen in practice. Other vulnerabilities in the same process can lead to the compromise of that buffer.

I think it's a good idea to be specific in what exactly the user would be protecting against by going off to use the `zeroize` crate.

Reply:
Also, I considered suggesting that there should be a note that says something like "You likely don't have to use this." I decided against that suggestion, though, when I considered that this is in the `hazardous` module, and it's fair to assume that the people using it know what they're doing.

Reply:
> I think it's a good idea to be specific about what exactly the user would be protecting against by going off to use the zeroize crate.

Very nice addition to the documentation. It really does make it much clearer and I feel like I'm able to much more quickly ascertain whether or not this is something I _do_ care about. Actually makes a world of difference.

Reply:
> Also, I considered suggesting that there should be a note that says something like "You likely don't have to use this." I decided against that suggestion, though, when I considered that this is in the hazardous module, and it's fair to assume that the people using it know what they're doing.

Like i mentioned in #178, this part I had forgotten about. I completely agree with your reasoning.